### PR TITLE
feat: add HA configuration for vcluster

### PR DESCRIPTION
**Pull request by pratikjagrut**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind feature

**What does this pull request do?**
This pull request adds HA configuration for vcluster.

- The first pod-anti-affinity rule says that the scheduler should try to avoid scheduling the Pod onto a node with one or more Pods with the label app=vcluster.

- The second pod-anti-affinity rule says that the scheduler should try to avoid scheduling the Pod onto a node in the same zone as one or more Pods with the label app=vcluster.

- The pod Disruption Budget makes sure that there is a maximum 1 pod unavailable at a time.

**Please provide a short message that should be published in the vcluster release notes**
Added an HA configuration for vcluster.

### CODE REVIEW

**Review by matskiv — CHANGES_REQUESTED** [review]

File: charts/k8s/values.yaml [hunk 1]
```diff
@@ -57,7 +58,7 @@ sync:
     # if true, vcluster will run with a scheduler and node changes are possible
     # from within the virtual cluster. This is useful if you would like to
     # taint, drain and label nodes from within the virtual cluster
-    enableScheduler: false
+    enableScheduler: true
```
[the remark this answers is not in the document]
Comment (pratikjagrut) [reply charts/k8s/values.yaml:61]: oh yes
testing part

File: charts/k8s/templates/api-deployment.yaml [hunk 2]
```diff
@@ -38,7 +38,39 @@ spec:
         release: {{ .Release.Name }}
     spec:
       terminationGracePeriodSeconds: 10
-      {{- if .Values.api.affinity }}
+      {{- if .Values.enableHA }}
```
[the remark this answers is not in the document]
Comment (pratikjagrut) [reply charts/k8s/templates/api-deployment.yaml:41]: Fabian said to enable it only if enableHA is enabled.

File: charts/k8s/values.yaml [hunk 3]
```diff
@@ -4,8 +4,9 @@ defaultImageRegistry: ""
 
 # If the control plane is deployed in high availability mode
 # Make sure to scale up the syncer.replicas, etcd.replicas, api.replicas & controller.replicas
-enableHA: false
-
+enableHA: true
```
[the remark this answers is not in the document]
Comment (pratikjagrut) [reply charts/k8s/values.yaml:7]: test change, reverting it back.

File: charts/k8s/templates/api-deployment.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (matskiv) [reply charts/k8s/templates/api-deployment.yaml:41]: Okay, but I would still reverse the condition order and first check if user set `.Values.api.affinity` to prefer that over our definition (with enableHA check).

**Review by matskiv — APPROVED** [review]
LGTM :) :

**Review by FabianKramm — APPROVED** [review]
@pratikjagrut thanks for the PR! LGTM!

**Pull request merged**